# Training Overview

Welcome to the **UPLB-IRRI Internship/Training: Basic Bioinformatics Course** under the UPLB and IRRI partnership. This training is designed to equip participants with the basic bioinformatics skills necessary for performing bioinformatics analysis.

## Objective

The general objective of the training is to:

- Provide participants with the necessary **bioinformatics** skills for data manipulation, analysis, and the use of genomics-related bioinformatics software.
- Specifically, the course will focus on:
  - Training participants to use the **Linux** operating system. Performing **Linux operations** effectively.
  - Gaining hands-on experience with **bioinformatics** tools and techniques.

## Format

This is a **hybrid training setup**.

---

# Course Outline

## Training 1: Sequence to Variants

### Module 1: Introduction and Fundamentals
**Objective**: To introduce the Linux operating system and basic shell commands to equip students for command-line data analysis.

- Introduction to Linux
- Using WSL on Windows and installation guide
- Linux file system, hard and soft links, working directory
- Concept of shell, Linux processes, Environment variables, PATH variables, Shell expansion, Scripts
- Using Bash to manipulate common data formats (FASTA, SAM, VCF)

### Module 2: NGS Data Analysis for Genotyping Purposes
**Objective**: To introduce the GATK workflow for variant calling and apply it to identify and characterize genetic variants in rice.

- Introduction to Second Generation Sequencing
- Variant Calling Pipeline
- Exploring VCF data, filtering, and basic statistics

---

## Training 2: GWAS and Post-GWAS

### Module 3: Basic GWAS Tutorial and Introduction to Internship Special Project
**Objective**: Introduction to GWAS methodology and acquiring practical skills in GWAS analysis of rice datasets.

- Population Structure and Rice Diversity
- Introduction to GWAS
- Interpreting GWAS outputs: Manhattan and QQ plots
- Hands-on: Population structure
- Hands-on: GWAS using TASSEL

### Module 4: Post-GWAS using Rice SNP-Seek, Rice Galaxy, and RicePilaf. Introduction to Special Project
**Objective**: Learn practical skills in interpreting GWAS output, integrating with other omics data, and identification of candidate genes.

- Significance threshold. Effective number of independent tests
- Linkage disequilibrium clumping
- Interrogating genomic regions with SNP-Seek and CropGalaxy
- Post-GWAS analysis using RicePilaf

### Module 5: Special Project
**Objective**: To process, analyze, and interpret datasets using the skillsets learned from the training.
